### ISSUE TITLE
VI Keybindings

### ISSUE THREAD

**Issue by Nomadluap** [mattgodbolt/compiler-explorer#550]
I've managed to condition myself to not be able to write code without vi kebindings. It would be neat if there was some way to implement vi keybindings on the editor component.

**Comment by jaredwy** [comment]
Looks like the editor used doesn't support vim bindings yet. 
There is an open issues over there https://github.com/Microsoft/monaco-editor/issues/182

**Comment by mattgodbolt** [comment]
Like @jaredwy says, this is something we'd need the upstream editor to support.

**Comment by limslarmo** [comment]
Seems this gentleman named brijeshb42 has implemented VIM support https://github.com/microsoft/monaco-editor/issues/182#issuecomment-414124838.

It is currently well maintained and has been used in popular web apps: https://github.com/microsoft/monaco-editor/issues/182#issuecomment-441273387.
Can we integrate it?

**Comment by mattgodbolt** [comment]
We are currently behind on our Monaco installation due to a number of issues in our setup (mainly Webpack-related). We need to upgrade that before we can consider adding any new plugins.

**Comment by limslarmo** [comment]
It does seem Monaco-editor refactored their code a little bit. 
Monaco-vim demands monaco-editor ^0.14.0 so I tried to upgrade to that version the other day. I was able to get Vim key binding to work by default(see diff https://github.com/limslarmo/compiler-explorer). The user has to do ctrl+c to change to normal mode instead of esc tho.. which is a bit annoying.  

There are a few front end conflicts to fix before this feature is reviewable. I will find a weekend to finish that.
